src/main/java/com/fasterxml/jackson/dataformat/xml/XmlTagProcessors.java

```diff
@@ -0,0 +1,212 @@
+     */
+    public static XmlTagProcessor newPassthroughProcessor() {
+        return new PassthroughTagProcessor();
+    }
+
+    /**
+     * Generates a new tag processor that replaces all invalid characters in an
+     * XML tag name with a replacement string. This is a one-way processor, since
+     * there is no way to reverse this replacement step.
+     * <p>
+     * With this processor set (and {@code "_"} as the replacement string), a map
+     * with the keys {@code "123"} and {@code "$ I am <fancy>! &;"} will be written as:
+     *
+     * <pre>{@code
+     * <DTO>
+     *     <badMap>
+     *         <__I_am__fancy_____>xyz</__I_am__fancy_____>
+     *         <_23>bar</_23>
+     *     </badMap>
+     * </DTO>
+     * }</pre>
+     *
+     * @param replacement The replacement string to replace invalid characters with
+     *
+     * @since 2.14
+     */
+    public static XmlTagProcessor newReplacementProcessor(String replacement) {
+        return new ReplaceTagProcessor(replacement);
+    }
+
+    /**
+     * Equivalent to calling {@link #newReplacementProcessor(String)} with {@code "_"}
+     *
+     * @since 2.14
+     */
+    public static XmlTagProcessor newReplacementProcessor() {
+        return newReplacementProcessor("_");
+    }
+
+    /**
+     * Generates a new tag processor that escapes all tag names containing invalid
+     * characters with base64. Here the
+     * <a href="https://datatracker.ietf.org/doc/html/rfc4648#section-5">base64url</a>
+     * encoder and decoders are used. The {@code =} padding characters are
+     * always omitted.
+     * <p>
+     * With this processor set, a map with the keys {@code "123"} and
+     * {@code "$ I am <fancy>! &;"} will be written as:
+     *
+     * <pre>{@code
+     * <DTO>
+     *     <badMap>
+     *         <base64_tag_JCBJIGFtIDxmYW5jeT4hICY7>xyz</base64_tag_JCBJIGFtIDxmYW5jeT4hICY7>
+     *         <base64_tag_MTIz>bar</base64_tag_MTIz>
+     *     </badMap>
+     * </DTO>
+     * }</pre>
+     *
+     * @param prefix The prefix to use for tags that are escaped
+     *
+     * @since 2.14
+     */
+    public static XmlTagProcessor newBase64Processor(String prefix) {
+        return new Base64TagProcessor(prefix);
+    }
+
+    /**
+     * Equivalent to calling {@link #newBase64Processor(String)} with {@code "base64_tag_"}
+     *
+     * @since 2.14
+     */
+    public static XmlTagProcessor newBase64Processor() {
+        return newBase64Processor("base64_tag_");
+    }
+
+    /**
+     * Similar to {@link #newBase64Processor(String)}, however, tag names will
+     * <b>always</b> be escaped with base64. No magic prefix is required
+     * for this case, since adding one would be redundant because all tags will
+     * be base64 encoded.
+     */
+    public static XmlTagProcessor newAlwaysOnBase64Processor() {
+        return new AlwaysOnBase64TagProcessor();
+    }
+
+
+
+    private static class PassthroughTagProcessor implements XmlTagProcessor {
```

Comment:
Should this be made `public`? It could serve as the base implementation for custom processors.
(not sure if that makes sense, just wondering)